lib/crm/helpers/finders.rb

```diff
@@ -5,11 +5,20 @@ def self.included(base)
         base.extend self
       end
 
-      def find(id)
-        crm_object = crm_class.find(id)
-        return nil if crm_object.blank?
+      def find(id_or_ids)
+        case id_or_ids
+        when String
+          crm_object = crm_class.find(id_or_ids)
+          return nil if crm_object.blank?
 
-        new(crm_object.attributes)
+          new(crm_object.attributes)
+        when Array
+          crm_objects = Crm.find(id_or_ids).to_a
+          rejected_crm_objects = crm_objects.reject { |crm_object| crm_object.is_a?(crm_class) }
+          raise Errors::ResourceNotFound.new("Not of type #{crm_class.name}", rejected_crm_objects.map(&:id)) if rejected_crm_objects.present?
```

Comment:
I'd probably just stick with the standard `Errors::ResourceNotFound` message.

```ruby
raise Errors::ResourceNotFound.new('Items could not be found.', rejected_crm_objects.map(&:id)) if rejected_crm_objects.present?
```

Reason being that's what you're getting for doing something like `Crm::Account.find(contact_id)`, too. Well, you're getting something similar, that is...